### PR TITLE
Add function getMapColor function to block

### PR DESCRIPTION
**Pull request by Adrian8115**

### CODE REVIEW

**Comment by FrederoxDev** [comment]
Please make a note in the implementation stating that it is incomplete, and only a partial reimplementation

**Comment by Adrian8115** [comment]
comments have been added

File: AmethystAPI/src/minecraft/src/common/world/level/block/Block.cpp [hunk 1]
```diff
@@ -45,6 +45,7 @@ gsl::strict_not_null<const Block*> Block::setState(const BlockState& stateType,
 
 template gsl::strict_not_null<const Block*> Block::setState<int>(const BlockState& stateType, int value) const;
 
+// Just a partial reimplementation using `BlockLegacy::getMapColor`
```
Comment (FrederoxDev) [inline AmethystAPI/src/minecraft/src/common/world/level/block/Block.cpp:48]: For clarity also specify the version e.g. `// 1.20.71.1 - Partial reimplementation`

File: AmethystAPI/src/minecraft/src/common/world/level/block/Block.hpp [hunk 2]
```diff
@@ -35,6 +35,9 @@ class Block : public BlockComponentStorage {
     template <typename T>
     gsl::strict_not_null<const Block*> setState(const BlockState& stateType, T value) const;
 
+    // Just a partial reimplementation using `BlockLegacy::getMapColor`
+    // Needed because `BlockLegacy::getMapColor` is a protected function and the Block class
```
Comment (FrederoxDev) [inline AmethystAPI/src/minecraft/src/common/world/level/block/Block.hpp:39]: Simple is better, just do something like my other comment

**Review by FrederoxDev — CHANGES_REQUESTED** [review]

**Review by FrederoxDev — APPROVED** [review]

File: AmethystAPI/src/minecraft/src/common/world/level/block/Block.hpp (hunk 2 above)
Comment (Adrian8115) [reply AmethystAPI/src/minecraft/src/common/world/level/block/Block.hpp:39]: this should be covered now?
I mean I simplfied it ig...

**Review by FrederoxDev — APPROVED** [review]

**Pull request merged**